test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -2153,17 +2063,8 @@ func TestPreScorePlugin(t *testing.T) {
 func TestPreEnqueuePlugin(t *testing.T) {
 	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodSchedulingReadiness, true)()
 
-	// Create a plugin registry for testing. Register only a filter plugin.
-	enqueuePlugin := &PreEnqueuePlugin{}
-	// Plumb a preFilterPlugin to verify if it's called or not.
-	preFilterPlugin := &PreFilterPlugin{}
-	registry, prof := initRegistryAndConfig(t, enqueuePlugin, preFilterPlugin)
-
-	// Create the API server and the scheduler with the test plugin set.
-	testCtx := initTestSchedulerForFrameworkTest(t, testutils.InitTestAPIServer(t, "enqueue-plugin", nil), 1,
-		scheduler.WithProfiles(prof),
-		scheduler.WithFrameworkOutOfTreeRegistry(registry))
-	defer testutils.CleanupTest(t, testCtx)
+	testContext := testutils.InitTestAPIServer(t, "enqueue-plugin", nil)
+	defer testutils.CleanupTest(t, testContext)
```

Comment:
Do all tests always call `testutils.CleanupTest`?

If yes, then the code it executed might fit into a `t.Cleanup` callback that gets registered inside `testutils.InitTestAPIServer`.

Reply:
Moved inside the `InitTestAPIServer`. PTAL.